### ISSUE TITLE
Throughput of Resteasy Reactive's significantly decreased in 2.12.0.Final

### ISSUE THREAD

**Issue by yazinnnn** [quarkusio/quarkus#27743]
### Describe the bug

when i use 2.11.3.Final, the endpoint's throughput would be 410k rps

```
wrk -t10 -c500 -d60s http://localhost:8080/bench
Running 1m test @ http://localhost:8080/bench
  10 threads and 500 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     2.90ms    4.52ms  94.42ms   87.86%
    Req/Sec    41.19k     5.32k   93.90k    72.64%
  24591994 requests in 1.00m, 1.86GB read
Requests/sec: 409195.71
Transfer/sec:     31.61MB
```

when upgrade to 2.12.0.Final, throughput would decrease to 46k rps

```
wrk -t10 -c500 -d60s http://localhost:8080/bench
Running 1m test @ http://localhost:8080/bench
  10 threads and 500 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    11.41ms    6.35ms  88.30ms   71.20%
    Req/Sec     4.55k   349.58     7.88k    71.64%
  2714402 requests in 1.00m, 209.68MB read
Requests/sec:  45169.21
Transfer/sec:      3.49MB

```

#### dependencies
```
    implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
    implementation("io.quarkus:quarkus-kotlin")
    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
    implementation("io.quarkus:quarkus-arc")
    implementation("io.quarkus:quarkus-resteasy-reactive")
```

#### test endpoint 
`fun uni(): Uni<String> = Uni.createFrom().item("uni")`

#### cpu 
intel 10400 6c12t

### Output of `uname -a` or `ver`

Linux yazi-m820 5.15.57-2-MANJARO #1 SMP PREEMPT Mon Jul 25 01:32:51 UTC 2022 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.2" 2022-01-18 LTS

### GraalVM version (if different from Java)

OpenJDK Runtime Environment GraalVM 22.0.0.2 (build 17.0.2+9-LTS)

### Quarkus version or git rev

[2.12.0.Final]

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gradle-7.4.2

**Comment by geoand** [comment]
Thanks for reporting.

That's a pretty extrordinary regression, I'll look into it ASAP

**Comment by geoand** [comment]
I can confirm this, we have a massive performance drop.

**Comment by geoand** [comment]
This was actually picked up by us as well yesterday and fixed in https://github.com/quarkusio/quarkus/pull/27736. The fix will be part of `2.12.1.Final`

**Comment by yazinnnn** [comment]
> This was actually picked up by us as well yesterday and fixed in #27736. The fix will be part of `2.12.1.Final`

Thanks a lot！

**Comment by geoand** [comment]
💪🏼
